skada/transformers.py

```diff
@@ -0,0 +1,66 @@
+# Author: Remi Flamary <remi.flamary@polytechnique.edu>
+#
+# License: BSD 3-Clause
+
+from sklearn.utils import check_random_state
+
+from .base import BaseAdapter
+from .utils import check_X_y_domain
+
+
+class SubsampleTransformer(BaseAdapter):
+    """Transformer that subsamples the data.
+
+    This transformer is useful to speed up computations when the data is too
+    large. It randomly selects a subset of the data to work with during training
+    but does not change the data during testing.
+
+    .. note::
+        This transformer should not be used as the last step of a pipeline
+        because it returns non standard output.
+
+    Parameters
+    ----------
+    n_subsample : int
+        Number of samples to keep (keep all if data smaller).
+    random_state : int, RandomState instance or None, default=None
+        Controls the random resampling of the data.
+    """
+
+    def __init__(self, n_subsample, random_state=None):
+        self.n_subsample = n_subsample
+        self.random_state = random_state
+
+    def _pack_params(self, idx, **params):
+        return {
+            k: (v[idx] if idx is not None else v)
+            for k, v in params.items()
+            if v is not None
+        }
+
+    def fit_transform(self, X, y=None, *, sample_domain=None, sample_weight=None):
+        """Fit and transform the data."""
+        X, y, sample_domain2 = check_X_y_domain(X, y, sample_domain)
+
+        self.rng_ = check_random_state(self.random_state)
+
+        if self.n_subsample >= X.shape[0]:
+            return (
+                X,
+                y,
+                self._pack_params(
+                    None, sample_domain=sample_domain, sample_weight=sample_weight
+                ),
+            )
+
+        idx = self.rng_.choice(X.shape[0], self.n_subsample, replace=False)
```

Comment:
In practical scenarios, do we need the sampling to be done per-domain? 🤔 It might be the case that the result of sampler omits domains with few(-er) entries.

Reply:
In this case do we create a DomainStratifiedSampleTransformer or do we add a parameter?  In any Case we should probably use sklearn or skada Splitters here

Reply:
I'm always for having separate classes if the use case is clear. Much easier to just open the list of things to spot what you're looking for! `DomainStratifiedSampleTransformer` sounds great.